cmd/shell.go

```diff
@@ -158,6 +158,27 @@ func (as *AppShell) Run() {
 	FailOnError(err, "Failed to run application")
 }
 
+// Clock functions similarly to clock.Default(), but the returned value can be
+// changed using the FAKECLOCK environment variable.
+//
+// The FAKECLOCK env var is in the time.UnixDate format, returned by `date -d`.
+func Clock() clock.Clock {
+	cl := clock.Default()
+	if tgt := os.Getenv("FAKECLOCK"); tgt != "" {
+		targetTime, err := time.Parse(tgt, time.UnixDate)
+		if err != nil {
+			fmt.Fprintf("cmd.Clock: bad format for FAKECLOCK: %v\n", err)
```

Comment:
confused, I am: `fmt.Fprintf`'s first argument is supposed to be an io.Writer`, I thought? Wouldn't this be `fmt.Printf`?